### DISCUSSION TITLE
Airburst Weapon & shield armor paradox

### DISCUSSION THREAD

**Discussion by 8abs** [Phobos-developers/Phobos#1017] [Ideas and feedback]
### First check

- [X] The issue happens on the **latest official** version of Phobos and wasn't fixed yet.
- [X] I agree to elaborate the details if requested and provide thorough testing if the bugfix is implemented.
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search and read the issue list to find a similar issue and didn't find it.
- [X] I have attached as much information as possible *(screenshots, gifs, videos, debug and exception logs, etc)*.

### Description

I am using build27.
If I set a weapon's warhead versus some shield armortype equals 0%, and this warhead contains EMP or AttachEffect.SpeedMultiplier, it would function well.
But if I set a weapon's warhead versus some shield armortype equals 100%, and at the same time set its airburst weapon's warhead versus the same shield armortype equals 0%. The first weapon has no EMP or AttachEffect.SpeedMultiplier, but its airburst weapon has one of EMP and AttachEffect.SpeedMultiplier, it turns out that the shielded unit would be EMPed or slowed down.

I checked. The shield is PassPercent=0% & AbsorbPercent=100%. So there's no chance any damage leak to itself. I did this experiment under serious controlled variable method and did it again and again. So I'm hardly wrong.

BTW, the shield in my test is from a warhead using"Shield.AttachTypes=" to a valid unit.

### Phobos Version

build27

### Steps to reproduce

Do as above.  I save the regedit part and unimportant parts of below items. The shield in my test is from a warhead using "Shield.AttachTypes=" to a valid unit.
[SHIELD1]
Armor=shield1
...

[Weapon1]
...
Projectile=Projectile1
Warhead=Warhead1

[Warhead1]
...
Versus.shieldarmor1=100%

[Projectile1]
...
AirBurstWeapon=Weapon2

[Weapon2]
...
Warhead=Warhead2

[Warhead2]
...
Versus.shieldarmor1=0%
AttachEffect.SpeedMultiplier=0.1
AttachEffect.Duration=-1

### Expected behaviour

UNIT1 shouldn't be slowed down

### Actual behaviour

UNIT1 did slowed down

**Comment by 8abs** [comment]
Addition: I didn't use "Shield.Penetrate=yes"

**Comment by Starkku** [comment]
Shields do not protect from Warhead effects, including Ares-introduced ones. Only damage. This behaviour cannot be easily changed either.

**Comment by 8abs** [comment]
> Shields do not protect from Warhead effects, including Ares-introduced ones. Only damage. This behaviour cannot be easily changed either.

Jesus, I always thought I was wrong somewhere. Is shield going to be able to defense attach-effects in the future?

**Comment by Starkku** [comment]
> > Shields do not protect from Warhead effects, including Ares-introduced ones. Only damage. This behaviour cannot be easily changed either.
> 
> Jesus, I always thought I was wrong somewhere. Is shield going to be able to defense attach-effects in the future?

Unlikely.
